### ISSUE TITLE
Possible integrity check improvement

### ISSUE THREAD

**Issue by joseph-robertson** [NREL/OpenStudio-BuildStock#241]
I feel like there is still a scenario that goes uncheck.

Consider this example in options_lookup.tsv:

parameter | option | measure
-- | -- | --
Cooking Range | Gas, 80% Usage | ResidentialApplianceCookingRange
Cooking Range | Gas, 100% Usage | ResidentialApplianceCookingRange
Cooking Range | Gas, 120% Usage | ResidentialApplianceCookingRange
Cooking Range | None |  
Cooking Range Schedule | California Average | ResidentialApplianceCookingRange
Cooking Range Schedule | None |  

Say Cooking Range.tsv has an option for None but Cooking Range Schedule.tsv does not. Then BuildExistingModel can fail on building the model since less than all of the cooking range measure's arguments could exist. However, the integrity check passes because None isn't included in these keys: https://github.com/NREL/OpenStudio-BuildStock/blob/master/Rakefile#L337. It appears to only check combinations of non-"None" options.

**Comment by shorowit** [comment]
The solution is to 1) have Cooking Range Schedule dependent on Cooking Range and 2) have probability distributions that ensure the two None options are only ever called together (or not called together). But that would be difficult for the integrity checks to catch since it requires some sampling of the probability distributions. 

The basic problem stems from the fact that different options within a parameter can reference different measures (California Average references ResidentialApplianceCookingRange but None references no measure). One way to address this would be to require that all options in a parameter reference the same measures (and arguments). This would be easy to add to our integrity checks. But it would require some work to align our options_lookup.tsv with this approach -- HVAC and perhaps geometry might be challenging.

Using this solution, the cooking range example would become:

parameter | option | measure | arg1
-- | -- | -- | --
Cooking   Range | Gas, 80% Usage | ResidentialApplianceCookingRange | Mult=0.8
Cooking   Range | Gas, 100% Usage | ResidentialApplianceCookingRange | Mult=1.0
Cooking   Range | Gas, 120% Usage | ResidentialApplianceCookingRange | Mult=1.2
Cooking   Range | None | ResidentialApplianceCookingRange | Mult=0.0
Cooking   Range Schedule | California Average | ResidentialApplianceCookingRange | Sched=…

In other words, the "None" options for each parameter would still call the corresponding measure, but their arguments would tell the measure that it should bail out and not actually add said building component to the model.
